packages/fast-breakpoint-tracker-react/src/breakpoint-tracker.tsx

```diff
@@ -0,0 +1,101 @@
+import * as React from "react";
+import { canUseDOM } from "exenv-es6";
+import { throttle } from "lodash-es";
+import { Breakpoints, defaultBreakpoints, identifyBreakpoint } from "./breakpoints";
+
+export interface BreakpointTrackerProps {
+    /**
+     * The render method
+     */
+    render: (activeBreakpoint: keyof Breakpoints | void) => React.ReactNode;
+}
+
+export interface BreakpointTrackerState {
+    activeBreakpoint: keyof Breakpoints | void;
+}
+
+export class BreakpointTracker extends React.Component<BreakpointTrackerProps, BreakpointTrackerState> {
+    /**
+     * The current breakpoint.
+     */
+    public static breakpoints: Breakpoints = defaultBreakpoints;
+
+    /**
+     * Track if we have an open animation frame request
+     */
+    private openRequestAnimationFrame: boolean;
+
+    /**
+     * Constructor for the BreakpointTracker component.
+     */
+    constructor(props: BreakpointTrackerProps) {
+        super(props);
+
+        this.state = {
+            activeBreakpoint: null
+        };
+    }
+
+    /**
+     * React life-cycle method
+     */
+    public componentDidMount(): void {
+        if (canUseDOM()) {
+            const initialBreakpoint: keyof Breakpoints | void = identifyBreakpoint(window.innerWidth, BreakpointTracker.breakpoints);
+
+            this.setState({
+                activeBreakpoint: initialBreakpoint
+            });
+
+            window.addEventListener("resize", this.requestFrame);
+        }
+    }
+
+    /**
+     * React life-cycle method
+     */
+    public componentDidUpdate(): void {
+        if (canUseDOM()) {
```

Comment:
Why does this need to update the breakpoint when the component updates? Shouldn't it just update the breakpoint when the window resizes?

Reply:
This doesn't belong here...